### ISSUE TITLE
TreeRangeSet still down't work in GWT

### ISSUE THREAD

**Issue by anton-tregubov** [google/guava#3049]
`@GwtIncompatible("uses NavigableMap")
public class TreeRangeSet`

That means that users with  GWT 2.7.0+(https://github.com/gwtproject/gwt/issues/4236) unable to use TreeRangeSet. Because annotated files with GwtIncompatible are skip sources in guava-gwt. 

Please remove it or describe why it's not possible.

Also maybe you know easy WA to make TreeRangeSet usable?

**Comment by cpovirk** [comment]
Last time I looked into this, I spent a little time and then concluded:

> This is looking nontrivial, since our RangeSet implementations rely on our other NavigableSet implementations, which we haven't yet made available under GWT.

It would be nice to do someday, but I wouldn't expect it to become a priority, sorry :(

**Comment by anton-tregubov** [comment]
https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/TreeRangeSet.java
Use only NavigableMap. 

I actually don't understand your point about NavigableSet. And even if we accept fact that there are some guava implementations that are not available under GWT, i think this classes are marked with @ GwtIncompatible and unavailable with or without fixed TreeRangeSet.

**Comment by cpovirk** [comment]
Sorry, you're right, `NavigableMap`, not `NavigableSet`.

That includes `AbstractNavigableMap` and its dependencies, including `DescendingMap` and `NavigableKeySet`.

**Comment by nikhilbarar** [comment]
@cpovirk Can i send a PR for the changes if any required?

**Comment by cpovirk** [comment]
This has turned out to look more complicated than we'd hoped, but we could take a patch if you get it working.

**Comment by aepshteyn** [comment]
I agree with @anton-tregubov - there doesn't seem to be any good reason why the `RangeSet` _interface_ is marked `@GwtIncompatible`.  This needlessly prevents users from writing their own implementation that _is_ compatible.

**Comment by aepshteyn** [comment]
> [@cpovirk](https://github.com/cpovirk) Can i send a PR for the changes if any required?

@nikhilbarar did you have any luck with this?
